### PR TITLE
Fix colouring issue when filtering

### PR DESCRIPTION
**Pull request by ekowalsk**
If applied, this pr will fix an issue with colouring when using filter on Accounts, Contacts and Bounties. Now it does not preserve the coloring of even and odd rows after applying a filter:
Currently: 
image
If applied: 
image
Also, the version of styled-components in resolutions was upgraded to match version in devDependencies

### CODE REVIEW

File: packages/test-support/package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
   "devDependencies": {
     "@polkadot/types-support": "^6.0.5",
     "@testing-library/jest-dom": "^5.14.1",
+    "jest-styled-components": "^7.0.5",
```
Comment (krzysztof-jelski) [inline packages/test-support/package.json:14]: I believe we haven't used it in the end. Can be safely removed.

File: packages/page-bounties/src/Bounties.tsx [hunk 2]
```diff
@@ -37,17 +37,17 @@ function Bounties ({ className }: Props): React.ReactElement {
         className='bounties-table-wrapper'
         empty={bounties && t<string>('No open bounties')}
         header={headerRef.current}
+        withCollapsibleRows={true}
```
Comment (MiZiet) [inline packages/page-bounties/src/Bounties.tsx:40]:
```suggestion
        withCollapsibleRows
```

File: packages/page-accounts/src/Accounts/index.tsx [hunk 3]
```diff
@@ -279,13 +279,10 @@ function Overview ({ className = '', onStatusChange }: Props): React.ReactElemen
       <Table
         empty={!isLoading && sortedAccounts && t<string>("You don't have any accounts. Some features are currently hidden and will only become available once you have accounts.")}
         header={header.current}
+        withCollapsibleRows={true}
```
Comment (MiZiet) [inline packages/page-accounts/src/Accounts/index.tsx:282]:
```suggestion
        withCollapsibleRows
```

File: packages/test-support/package.json (hunk 1 above)
Comment (MiZiet) [inline packages/test-support/package.json:14]: do we use it?

File: packages/page-addresses/src/Contacts/index.tsx [hunk 4]
```diff
@@ -75,12 +75,12 @@ function Overview ({ className = '', onStatusChange }: Props): React.ReactElemen
       <Table
         empty={!isLoading && sortedAddresses && t<string>('no addresses saved yet, add any existing address')}
         header={headerRef.current}
+        withCollapsibleRows={true}
```
[the remark this answers is not in the document]
Comment (ekowalsk) [reply packages/page-addresses/src/Contacts/index.tsx]: ✅

File: packages/page-accounts/src/Accounts/index.tsx (hunk 3 above)
Comment (ekowalsk) [reply packages/page-accounts/src/Accounts/index.tsx]: ✅

**Comment by ekowalsk** [comment]
The version of styled-components in resolutions was upgraded to match version in devDependencies

File: packages/react-components/src/Table/index.tsx [hunk 5]
```diff
@@ -104,6 +105,18 @@ export default React.memo(styled(Table)`
...
+      tr:nth-child(4n - 2),
+      tr:nth-child(4n - 3) {
+        background-color: var(--bg-table);
+      }
+
+      tr:nth-child(4n - 1),
+      tr:nth-child(4n) {
+        background-color: unset;
+      }
+    }
```
Comment (MiZiet) [inline packages/react-components/src/Table/index.tsx:119]:
```suggestion
  &.withCollapsibleRows tbody tr {
      background-color: unset;

      &:nth-child(4n - 2),
      &:nth-child(4n - 3) {
        background-color: var(--bg-table);
      }
    }
```

**Review by MiZiet — APPROVED** [review]

**Review by jacogr — APPROVED** [review]

**Pull request merged**